src/analytics/src/main/scala/osmesa/analytics/oneoffs/ChangesetMetadataUpdater.scala

```diff
@@ -80,7 +80,8 @@ object ChangesetMetadataUpdater
               import ss.implicits._
 
               val options = Map(
-                Source.BaseURI -> changesetSource.toString
+                Source.BaseURI -> changesetSource.toString,
+                Source.ProcessName -> "ChangesetStream"
```

Comment:
This doesn't achieve what you were hoping it did. `Source.ProcessName` is only used by _streaming_ sources (in `commit` after processing batches); this is a regular source.

Instead, we should have something that can be executed after `forEachPartition` completes to update the checkpoints appropriately.

Reply:
It seems like this (and the mirrored comment below) can just be omitted?  I've been really confused while trying to figure out how the various classes know where to look for their current sequence number.  I'm also not clear why this isn't a streaming source.  Do we ever `spark.readStream` anywhere?  The nature of the sources supplied by VP is a continuing source of confusion that I would love to clear up.

Reply:
`StreamingChangesetMetaUpdater` uses `spark.readStream`. `readStream` creates a streaming source (using the "MicroBatchReader" implementation, producing an unending stream of batched updates), `read` creates a non-streaming source (producing a single `DataFrame` with a finite range of updates).

Start sequence calculation / recovery is handled in 2 places (which could probably be merged, causing `Source.ProcessName` to be used in non-streaming contexts, although only streaming sources will **update** the checkpoint):

* [`ReplicationStreamMicroBatchReader`](https://github.com/geotrellis/vectorpipe/blob/cbeb773ed9dc6bf211221611f136047c435451e4/src/main/scala/vectorpipe/sources/ReplicationStreamMicroBatchReader.scala#L42-L49) -- when `readStream` is used. Resolution order is: 1) `checkpoints` table, 2) `Source.StartSequence` option
* [`ReplicationReader`](https://github.com/geotrellis/vectorpipe/blob/cbeb773ed9dc6bf211221611f136047c435451e4/src/main/scala/vectorpipe/sources/ReplicationReader.scala#L34-L40) -- when `read` is used. Resolution order is: 1) `Source.StartSequence` option, 2) `getCurrentSequence`

For the `ReplicationStreamMicroBatchReader`, the 3rd resolver **should** be `getCurrentSequence`.

Reply:
I'll throw up a PR to fix this in VP.